src/common/fam_libfabric.cpp

```diff
@@ -576,8 +576,18 @@ int fabric_completion_wait(Fam_Context *famCtx, fi_context *ctx) {
     int timeout_wait_retry_cnt = 0;
 
     do {
+        if (  (ctx->internal[1] ==  (void *)FAM_REQ_COMPLETED) ) {
```

Comment:
request might be completed successfully or completed with err. Need to raise an exception for request completed with error. You might have to save the provider error state and use it while raising exception.